src/components/data/ScrollContext.js

```diff
@@ -0,0 +1,10 @@
+import { createContext } from 'preact';
+
+/**
+ * @typedef ScrollInfo
+ * @prop {import('preact').RefObject<HTMLElement>} scrollRef
+ */
+
+const ScrollContext = createContext(/** @type {ScrollInfo} */ ({}));
+
+export default ScrollContext;
```

Comment:
In these changes, contexts are default exports. This matches patterns I saw in other libraries, but I'm totally flexible.